### PR TITLE
Saved images

### PR DESCRIPTION
**Pull request by christocarr**
- add save button to modal to allow users to toggle a saved image
- add a custom hook to save image to local storage
- display saved images on saved page
- change modal to use react portal

### CODE REVIEW

File: src/components/Modal/Modal.js [hunk 1]
```diff
@@ -1,57 +1,73 @@
...
   useEffect(() => {
-    const dateNow = Date.now();
-    const dateCreated = Date.parse(location.state.createdAt);
+    if (isOpen) {
+      const dateNow = Date.now();
+      const dateCreated = Date.parse(content.created_at);
 
-    let diff = Math.abs(dateNow - dateCreated) / 1000;
-    const hours = Math.floor(diff / 3600) % 24;
+      let diff = Math.abs(dateNow - dateCreated) / 1000;
```
Comment (yoloOnTheBattlefield) [inline src/components/Modal/Modal.js]: any reason why you need a let here?

File: src/pages/Saved/Saved.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import useLocalStorage from 'utils/useLocalStorage';
+import SaveImage from 'utils/SaveImage';
```
Comment (yoloOnTheBattlefield) [inline src/pages/Saved/Saved.js]: any reason why this PascalCase?

File: src/pages/SearchResults/SearchResults.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import InfiniteLoader from 'react-infinite-loader';
+import SaveImage from 'utils/SaveImage';
```
Comment (yoloOnTheBattlefield) [inline src/pages/SearchResults/SearchResults.js]: same here?

File: src/utils/useLocalStorage.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+//         localStorage.setItem(key, serializedNewVal);
+//       }
+//       setValue(newValue ?? initialValue);
+//     },
+//     [initialValue, serializedInitialVal, key]
+//   );
+//   return [value, updatedSetValue];
+// }
+
+// export default useLocalStorage;
```
Comment (yoloOnTheBattlefield) [inline src/utils/useLocalStorage.js]: no need for commented out code

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/pages/Saved/Saved.js (hunk 2 above)
Comment (christocarr) [reply src/pages/Saved/Saved.js]: Screen Shot 2021-08-16 at 16 01 59

**Review by yoloOnTheBattlefield — APPROVED** [review]

**Pull request merged**